demo/scenes/entities/units/soldier/states/attack.gd

```diff
@@ -0,0 +1,10 @@
+extends State
+
+func enter() -> void:
```

Comment:
I added some `params` the enter method signature at the state, so you need to update it here:
 
`func enter(params: Dictionaty) -> void:`